python/Ganga/Utility/execute.py

```diff
@@ -230,10 +228,10 @@ def execute(command,
 
     if update_env:
         env_output_key = 'env_output'
-        update_env_thread = update_thread(env_file_pipes, thread_output, env_output_key)
+        update_env_thread = update_thread(env_file_pipes, thread_output, env_output_key, require_output=True)
     if not shell:
         pkl_output_key = 'pkl_output'
-        update_pkl_thread = update_thread(pkl_file_pipes, thread_output, pkl_output_key)
+        update_pkl_thread = update_thread(pkl_file_pipes, thread_output, pkl_output_key, require_output=False)
```

Comment:
I'm confused now. If you expect some output you should ask for it. What is the situation where there is output through this stream where it's not been explicitly asked for? This seems like a strange mis-use of the API to me.